### PR TITLE
Refactoring issue submission and log files reader

### PR DESCRIPTION
**Pull request by ratoaq2**
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read [contribution guide](https://github.com/PyMedusa/SickRage/blob/master/.github/CONTRIBUTING.md)
- [x] Refactored issue submission logic to a separate module
- [x] Log reading improvement to be efficient and to not be memory hungry.
- [x] Better gist logs
- [x] Flake8 fixes on the modified files
- [x] Log reader tests
- [x] issue submitter tests
- [x] Improved similar issue submission to be more generic: if title is 90% similar to an existing issue, it will comment the issue instead of creating a new one. Also restricting to the last 90 days to avoid full issue scan on github.
- [x] Added Period filter to log view

### CODE REVIEW

**Comment by fernandog** [comment]
@ratoaq2 is it possible to have a "Real-time" log in view log - as if we are in console?

**Comment by ratoaq2** [comment]
It's possible with some help from a frontend dev

**Comment by fernandog** [comment]
@ratoaq2 we can try later. as soon you say this PR is done will test it. and then we can try "live log"

**Comment by ratoaq2** [comment]
Probably live logs should go in a PR after this one.

I'm still trying something in this PR... Search logs is always slow and when you specify a lot of criteria because it has to search more files. It would be nice to have another filter criteria to restrict the timestamp: 1h, 1day, 1week... something like that

**Comment by fernandog** [comment]
@ratoaq2 

doesnt show traceback anymore

image

**Comment by fernandog** [comment]
@ratoaq2 also i think when develop=1 it sholdn't check UNSUPPORTED_VERSION. 

Im running my own branch so ofc its not updated

**Comment by OmgImAlexis** [comment]
If you wanted live logs we could setup an API endpoint for it and use that with polling for now. I'd need to talk to @ratoaq2 about naming conventions, etc. for it though.

**Comment by fernandog** [comment]
Its not showing provider name and thread name:

Now:
```
2016-08-22 01:33:14 Unable to connect to provider
 ```

Before:
```
SEARCHQUEUE-BACKLOG-121361 :: [Rarbg] :: Unable to connect to provider
SEARCHQUEUE-DAILY-SEARCH :: [Rarbg] :: Unable to connect to provider
```

File: sickbeard/issuesubmitter.py [hunk 1]
```diff
@@ -28,17 +27,6 @@ class IssueSubmitter(object):
...
-        # [APP SUBMITTED]:
-        #   'ascii' codec can't encode characters in position 00-00: ordinal not in range(128)
-        # [APP SUBMITTED]:
-        #   'charmap' codec can't decode byte 0x00 in position 00: character maps to <undefined>
-        re.compile(r'.* codec can\'t .*code .* in position .*:'),
-
-        # [APP SUBMITTED]: not well-formed (invalid token): line 0, column 0
-        re.compile(r'.* not well-formed \(invalid token\): line .* column .*'),
-    }
-
```
Comment (fernandog) [inline sickbeard/issuesubmitter.py]: @ratoaq2 what is the alternative for this?

**Comment by fernandog** [comment]
not sure about this: 
` Also restricting to the last 90 days to avoid full issue scan on github`

if the issue is older and has a fix/explanation how to fix, we will need to maually scan older issues so

**Comment by ratoaq2** [comment]
Have an open ended search for repo issues might be problematic in a near future. I'll do some tests calling github rest api to see what's reasonable. One initial idea would be to have a label for APP Submitted bugs and then we could extend the period to check similar bugs. IMO we should have used labels from the start instead of the preformatted title, since they are easy to filter using web and rest api.

**Comment by ratoaq2** [comment]
Still... I don't know if permissions might be a problem since only we are allowed to label issues

**Comment by fernandog** [comment]
@OmgImAlexis  got this after @ratoaq2 fixed TORNADO logging.

I know donate.jpg doesnt exist. So we must remove it where is calling.
The others maybe got removed too?

```
2016-08-23 17:25:02 WARNING  TORNADO :: [b2476ac] 404 GET /js/core.min.js?12168 (143.107.176.135) 3.16ms
2016-08-23 17:25:01 WARNING  TORNADO :: [b2476ac] 404 GET /images/donate.jpg (143.107.176.135) 2.94ms
2016-08-23 17:25:01 WARNING  TORNADO :: [b2476ac] 404 GET /js/core.min.js?12168 (143.107.176.135) 5.72ms
```

**Comment by fernandog** [comment]
@OmgImAlexis @rarbg 

about real time logger, i would have button to popup a small window that fetches `/?cmd=logs` each 5 seconds and show it in reverse order (like in ssh/cmd)

**Comment by p0psicles** [comment]
Please scope this down. I believe we started this PR as for issue submissions? Real time log, is a new feature, new PR.

**Comment by fernandog** [comment]
@p0psicles its just a discussion.

**Comment by fernandog** [comment]
im not going to develop that. I'm only saying my ideas. 
will discuss this with them in slack ok?

**Comment by fernandog** [comment]
@medariox @ratoaq2 downloading a medusa `.zip` from github and running in windows with `python.exe` 

it shows:
`INFO     MAIN :: [] Checking for scene exception updates for AniDB`

in linux with `git clone` installation:
`INFO     MAIN :: [9dc5945] Checking for scene exception updates for AniDB`

i remember we check if current commit hash exists before adding it to log

**Comment by fernandog** [comment]
@ratoaq2 can you rebase please?

**Comment by fernandog** [comment]
@ratoaq2 rebase please?

**Comment by fernandog** [comment]
Approved

**Pull request merged**